platform/disk/root_device_partitioner.go

```diff
@@ -170,17 +169,26 @@ func (p rootDevicePartitioner) GetPartitions(devicePath string) (
 			return partitions, deviceFullSizeInBytes, bosherr.WrapErrorf(err, "Parsing existing partitions of `%s'", devicePath)
 		}
 
+		partitionType := PartitionTypeUnknown
+		if partitionInfo[4] == "ext4" || partitionInfo[4] == "xfs" {
+			partitionType = PartitionTypeLinux
+		} else if partitionInfo[4] == "linux-swap(v1)" {
+			partitionType = PartitionTypeSwap
+		} else if partitionInfo[4] == "fat16" {
+			partitionType = PartitionTypeEFI
+		}
+
```

Comment:
I couldn't follow here whether this is an additional improvement or related to the EFI based stemcell change. The same with the new if in line 107 to 109. Maybe an additional comment could help here.

Reply:
I added comments. Let me know if it is helpful. It is used to determine if we are using EFI to check 2nd partition instead of 1st. We only assumed 1st before.

Reply:
@lnguyen Thanks! It helps me at least.